src/preamble.js

```diff
@@ -2114,6 +2121,9 @@ function integrateWasmJS(Module) {
     wasmBinaryFile = Module['locateFile'](wasmBinaryFile);
     asmjsCodeFile = Module['locateFile'](asmjsCodeFile);
   }
```

Comment:
so the goal is to support both locateFile and scriptDirectory at the same time (otherwise an else here could make sense)?

Reply:
Yes, that's the idea. The existing ways to form a URL to load will all be treated with respect to `scriptDirectory` if the specified URLs are relative.